File: 3541-report-spam-message/3541-report-spam-message.cpp
class Solution {
public:
    bool reportSpam(vector<string>& message, vector<string>& bannedWords) {
        unordered_set<string> set;
        for(const string& s: bannedWords){
            set.insert(s);
        }
        int c=0;
        for(string const& s:message){
            if(set.find(s)!=set.end()) c++;
            if(c==2) return true;
        }
        return false;
    }
};

File: 3541-report-spam-message/3541-report-spam-message.java
class Solution {
    public boolean reportSpam(String[] message, String[] bannedWords) {
        HashSet<String> set=new HashSet<>();
        for(String s:bannedWords){
            set.add(s);
        }
        int c=0;
        for(String s:message){
            if(set.contains(s)) c++;
            if(c==2) return true;
        }
        return false;
    }
}